clients/cli/src/sync_metadata_to_token2022.rs

```diff
@@ -0,0 +1,156 @@
+use {
+    crate::{
+        common::{parse_pubkey, process_transaction},
+        config::Config,
+        output::{format_output, println_display},
+        CommandResult,
+    },
+    clap::{ArgMatches, Args},
+    mpl_token_metadata::accounts::Metadata as MetaplexMetadata,
+    serde_derive::{Deserialize, Serialize},
+    serde_with::{serde_as, DisplayFromStr},
+    solana_cli_output::{display::writeln_name_value, QuietDisplay, VerboseDisplay},
+    solana_pubkey::Pubkey,
+    solana_remote_wallet::remote_wallet::RemoteWalletManager,
+    solana_signature::Signature,
+    solana_signer::Signer,
+    solana_transaction::Transaction,
+    spl_token_wrap::{
+        get_wrapped_mint_address, get_wrapped_mint_authority,
+        instruction::sync_metadata_to_token_2022,
+    },
+    std::{
+        fmt::{Display, Formatter},
+        rc::Rc,
+    },
+};
+
+#[derive(Clone, Debug, Args)]
+pub struct SyncMetadataToToken2022Args {
+    /// The address of the unwrapped mint whose metadata will be synced from
+    #[clap(value_parser = parse_pubkey)]
+    pub unwrapped_mint: Pubkey,
+
+    /// Specify that the source metadata is from a `Metaplex` Token Metadata
+    /// account. The CLI will derive the PDA automatically.
+    #[clap(long)]
+    pub metaplex: bool,
```

Comment:
Rather than forcing people to specify this, it would be better to figure it out dynamically.

* If `metadata_account` is specified, use that.
* If the unwrapped mint is SPL-Token, then use metaplex.
* If the unwrapped mint is SPL-Token-2022, then check for the metadata pointer, and fall back to metaplex.